components/x-interaction/__tests__/registerComponent.test.js

```diff
@@ -0,0 +1,54 @@
+const {
+	registerComponent,
+	getComponentByName,
+	getComponent,
+	getComponentName
+} = require('../src/concerns/register-component')
+const { withActions } = require('../')
+describe('registerComponent', () => {
+	let name
+	let Component
+	beforeAll(() => {
+		name = 'testComponent'
+		Component = withActions({})(() => null)
+	})
+	it(`should register a component in registerComponent`, () => {
+		registerComponent(Component, name)
+
+		const actualComponent = getComponentByName(name)
+		expect(actualComponent).toBeTruthy()
+	})
+	it('should throw an error if the component has already been registered', () => {
+		expect(() => registerComponent(Component, name)).toThrow(
+			'x-interaction a component has already been registered under that name, please use another name.'
+		)
+	})
+	it('should throw an error if the component is not x-interaction wrapped', () => {
+		const unwrappedComponentName = 'unwrappedComponent'
+		const unwrappedComponent = { '._wraps': null }
```

Comment:
**nitpick**: it doesn't really matter because it's the absense of the property that's important for the test, but there's an extra `.` in the property name here:

```suggestion
		const unwrappedComponent = { _wraps: null }
```

Reply:
good spot, thank you!